src/pow.rs

```diff
@@ -270,3 +346,42 @@ fn count_leading_zeros(data: &[u8]) -> u8 {
 
     leading_zeros
 }
+
+/// The faucet will dynamically adjust the difficulty based on the
+/// current balance of the faucet to make it increasingly difficult
+/// to retrieve funds from the faucet. The actual equation for this
+/// is:
+///
+/// ```math
+/// f(x)=(M-m)(1-\log_{q}\frac{x}{b})+m
```

Comment:
Recommend that we including the clamping here, so the subsequent guarantee on the difficulty range is satisfied.

Reply:
I would also include the expected clamping points for clarity: we should get `f(x) = M` at `x = b`, and `f(x) = m` at `x = bq`.

Reply:
Updated the docstring.